clean up docs on actions, code examples, reference to hybrid (#4721)
Unity-Technologies/ml-agents: docs/Python-API.md

  terminates the communication.
- **Behavior Specs : `env.behavior_specs`** Returns a Mapping of
  `BehaviorName` to `BehaviorSpec` objects (read only).

>   A `BehaviorSpec` contains information such as the observation shapes and the

Author: @vincentpierre @andrewcoh Can you review this part and make sure my changes are accurate?

Reviewer: Currently the behavior spec just contains those two things, left a suggestion to just explicitly say that.

>   A `BehaviorSpec` contains information such as the observation shapes and the

Reviewer: ```suggestion
  A `BehaviorSpec` contains the observation shapes and the
```

  number of agents is not guaranteed to remain constant during the simulation
  and it is not unusual to have either `DecisionSteps` or `TerminalSteps`
  contain no Agents at all.
- **Set Actions :`env.set_actions(behavior_name: str, action: ActionTuple)`** Sets
  the actions for a whole agent group. `action` is an `ActionTuple`, which

>   is made up of a 2D `np.array` of `dtype=np.int32` for discrete action, and
>   `dtype=np.float32` for continuous action case. The first dimension of `np.array`

Reviewer: ```suggestion
  is made up of a 2D `np.array` of `dtype=np.int32` for discrete actions, and
  `dtype=np.float32` for continuous actions. The first dimension of `np.array`
```

  for the corresponding Agent. This is used to track Agents across simulation
  steps.

> - `action_mask` is an optional list of two dimensional array of booleans. Only

Reviewer: ```suggestion
- `action_mask` is an optional list of two dimensional array of booleans which is only
```

  the last simulation step.
- `agent_id` is an int and an unique identifier for the corresponding Agent.

> - `action_mask` is an optional list of one dimensional array of booleans. Only

Reviewer: ```suggestion
- `action_mask` is an optional list of one dimensional array of booleans which is only
```
